add python3-transitions rosdep key

zhtroy opened the pull request:
<!-- Thank you for contributing a change to the rosdistro. There are two primary types of submissions.
Please select the appropriate template from below: ROSDEP_RULE_TEMPLATE or DOC_INDEX_TEMPLATE

If you're making a new release with bloom please use bloom to create the pull request automatically.
If you've already run the release bloom has a `--pull-request-only` option you can use.-->


<!-- ROSDEP_RULE_TEMPLATE: Submitter Please review the contributing guidelines: https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md -->

Please add the following dependency to the rosdep database.

## Package name:

python3-transitions

## Package Upstream Source:

https://github.com/pytransitions/transitions

## Purpose of using this:

after(including) ubuntu:bionic， python transitions is updated to python3 version, I'm using it in my project

Distro packaging links:

## Links to Distribution Packages

<!-- Replace the REQUIRED areas with the URL to the package.  For IF AVAILABLE areas, either put in the URL to the package or state 'not available'.
More info at https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#guidelines-for-rosdep-rules -->

- Debian: https://packages.debian.org/
  - https://packages.debian.org/search?keywords=python3-transitions&searchon=names&suite=all&section=all
- Ubuntu: https://packages.ubuntu.com/
   - https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=python3-transitions&searchon=names



<!-- DOC_INDEX_TEMPLATE: add package to rosdistro for documentation indexing -->

zhtroy committed:
add python3-transitions rosdep key

zhtroy committed:
add debian part

ivanpauno commented on rosdep/python.yaml:
```diff
@@ -5256,6 +5256,16 @@ python-transitions:
+        packages: [transitions]
+  ubuntu:
+    '*': [python3-transitions]
+    xenial: null
+    trusty: null
```
you could add the pip key for this two as well

zhtroy replied:
added in new commit

ivanpauno commented:
@zhtroy could you check the CI failures?

> rosdep/python.yaml#L5260
> Package 'python3-transitions' could not be found for debian buster on amd64

> rosdep/python.yaml#L5260
> [trailing-spaces] trailing spaces

zhtroy committed:
* remove trailing space
* change debian
* add xenial and trusty for ubuntu

zhtroy commented:
fixed CI problem

zhtroy committed:
use wildcard instead

zhtroy committed:
fixed indentation from 8 to 6

zhtroy committed:
Merge branch 'ros:master' into master

zhtroy committed:
fix CI problem
- 'trusty' should come before 'xenial'
- 'python-trep' should come before 'python3-transitions'

zhtroy commented:
> @zhtroy mind to check for package availability in Fedora, Gentoo, and RHEL? Thanks!

sorry, don't have much time

tfoote commented:
The CI fix has moved the `python-trep` incorrectly infront of `python-transitions`

zhtroy committed:
change python3-transitions entry ordering

zhtroy committed:
remove blank line
